docs/how-packages-and-artifacts-are-built.md

```diff
@@ -0,0 +1,114 @@
+# Overview
+
+DC/OS uses a custom packaging tool called [pkgpanda](../pkgpanda). This tool can read package metadata JSONs in a [package tree](../packages), verify the package dependency list is resolvable by walking through depedent packages, and then use the build script files accompanying the metadata to perform a docker-based build where the artifacts can be transferred out via a volume mount. Finally, these artifacts are packaged in XZ-compressed tar-balls.
```

Comment:
I wonder if we shouldn't add some background why we invented our own packaging system. I sometimes see people ranting on HN that we have `not-invented-here-syndrome`